tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/EmailAddressContact.scala

```diff
@@ -20,24 +23,25 @@ trait EmailAddressContactSearchEngine {
 }
 
 class InMemoryEmailAddressContactSearchEngine extends EmailAddressContactSearchEngine {
-  var emailList: Map[AccountId, Set[EmailAddressContact]] = Map()
+  val emailList: Multimap[AccountId, EmailAddressContact] = Multimaps.synchronizedSetMultimap(HashMultimap.create())
 
   override def index(accountId: AccountId, address: EmailAddressContact): Publisher[Unit] = {
-    var mailListPerAcc: Set[EmailAddressContact] = Set()
-    if (emailList.contains(accountId)) {
-      mailListPerAcc = emailList.apply(accountId)
-    }
-    mailListPerAcc = mailListPerAcc + address
-    emailList = emailList + (accountId -> mailListPerAcc)
-    SFlux.fromIterable(emailList).`then`()
+    SMono.fromCallable(() => {
+      if (emailList.containsKey(accountId)) {
+        emailList.get(accountId).add(address)
+      } else {
+        emailList.put(accountId, address)
+      }
+    }).`then`()
+
   }
 
   override def autoComplete(accountId: AccountId, part: String): Publisher[EmailAddressContact] = {
-    var addressList: Set[EmailAddressContact] = Set()
-    if (emailList.contains(accountId)) {
-      addressList = emailList.apply(accountId)
+    var addressList: Iterable[EmailAddressContact] = Set()
+    if (emailList.containsKey(accountId)) {
+      addressList = emailList.get(accountId).asScala
     }
     val fullList: SFlux[EmailAddressContact] = SFlux.fromIterable(addressList)
-    fullList.filter(e => e.address.contains(part))
+    fullList.filter(_.address.contains(part))
```

Comment:
1. Don't use `var`
2. get rid of the `if`
3. Wrap everything in a SMono. Just like the method above.

Reply:
Seems I used SFlux only without using SMono.just. Is it okie?